src/app_modes/topic_managment.rs

```diff
@@ -0,0 +1,365 @@
+    pub fn previous(&mut self) {
+        let i = match self.state.selected() {
+            Some(i) => {
+                if i == 0 {
+                    self.items.len()
+                } else {
+                    i - 1
+                }
+            }
+            None => 0,
+        };
+        self.state.select(Some(i));
+    }
+
+    pub fn add(&mut self, element: [String; 2]) {
+        self.items.push(element);
+    }
+
+    // Default to 0 if none is selected, the handling of empty vectors should be
+    // handled by the caller
+    pub fn pop(&mut self) -> [String; 2]{
+        let i = match self.state.selected() {
+            Some(i) => i,
+            None => 0,
+        };
+        self.items.remove(i)
+    }
+}
+
+pub struct TopicManager {
+    // Topic Manger loads the active and supported topics into two lists.
+    // The User can shift elements between available and selected topics.
+    // topics can only be present in on of the lists.
+
+    availible_topics: SelectableTopics,
+    selected_topics: SelectableTopics,
+    config: TermvizConfig,
+    selection_mode: bool,
+}
+
+impl TopicManager {
+    pub fn new(config: TermvizConfig) ->  TopicManager
+    {
+        let config = config.clone();
+
+        // Get all topics currently active in the config and sort them by topic type
+        let active_laser_topics: Vec<[String; 2]> = config.laser_topics.iter().map(
+            |i| {[i.topic.clone(),  "sensor_msgs/LaserScan".to_string()]}
+        ).collect();
+        let active_marker_array_topics: Vec<[String; 2]> = config.marker_array_topics.iter().map(
+            |i| {[i.topic.clone(),  "visualization_msgs/MarkerArray".to_string()]}
+        ).collect();
+        let active_marker_topics: Vec<[String; 2]> = config.marker_topics.iter().map(
+            |i| {[i.topic.clone(),  "visualization_msgs/Marker".to_string()]}
+        ).collect();
+        let active_pose_stamped_topics: Vec<[String; 2]> = config.pose_stamped_topics.iter().map(
+            |i| {[i.topic.clone(),  "geometry_msgs/PoseStamped".to_string()]}
+        ).collect();
+        let active_pose_array_topics: Vec<[String; 2]> = config.pose_array_topics.iter().map(
+            |i| {[i.topic.clone(),  "geometry_msgs/PoseArray".to_string()]}
+        ).collect();
+        let active_path_topics: Vec<[String; 2]> = config.path_topics.iter().map(
+            |i| {[i.topic.clone(),  "nav_msgs/Path".to_string()]}
+        ).collect();
+        // Collect them into a big list
+        let all_active_topics = [
+            active_laser_topics,
+            active_marker_array_topics,
+            active_marker_topics,
+            active_pose_stamped_topics,
+            active_pose_array_topics,
+            active_path_topics
+        ].concat();
+
+        
+        // We could get this from config, but would need some breaking changes in config
+        let supported_topic_types = vec![
```

Comment:
images are missing

Reply:
The image viewer will collect them automagically anyway

Reply:
what do you mean? It would be nice to also select images

Reply:
done